### PR TITLE
[MOS-972] Fixed missing contact entries when scrolling phonebook

### PR DESCRIPTION
**Pull request by Lefucjusz**
Fix of the issue that in some cases contact entry
would not be displayed on any page in case it
was on the boundary of two pages.

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [ ] Has unit tests if possible
- [ ] Has documentation updated
- [x] Has changelog entry added

### CODE REVIEW

File: module-apps/application-phonebook/widgets/PhonebookListView.hpp [hunk 1]
```diff
@@ -14,16 +14,20 @@ namespace gui
...
+                          std::uint32_t x,
+                          std::uint32_t y,
+                          std::uint32_t w,
+                          std::uint32_t h,
+                          std::shared_ptr<ListItemProvider> prov);
+
+      private:
         void addItemsOnPage() override;
         void addLabelMarker(PhonebookItem *item);
+        unsigned getSlotsLeft();
```
Comment (exeosdev) [inline module-apps/application-phonebook/widgets/PhonebookListView.hpp:28]: Maybe `uint32_t` or `size_t` to be more explicit?

File: module-gui/gui/widgets/ListViewEngine.hpp [hunk 2]
```diff
@@ -109,7 +109,7 @@ namespace gui
         [[nodiscard]] unsigned getFocusItemIndex();
 
         /// Total provider elements count
-        unsigned int elementsCount = listview::nPos;
+        unsigned elementsCount = listview::nPos;
```
Comment (exeosdev) [inline module-gui/gui/widgets/ListViewEngine.hpp:112]: `auto` would be even better here 🙂

**Review by dawidwojtas-mudita — APPROVED** [review]

File: module-apps/application-phonebook/widgets/PhonebookListView.hpp (hunk 1 above)
Comment (Lefucjusz) [reply module-apps/application-phonebook/widgets/PhonebookListView.hpp]: Changed to `std::size_t` ;)

File: module-gui/gui/widgets/ListViewEngine.hpp (hunk 2 above)
Comment (Lefucjusz) [reply module-gui/gui/widgets/ListViewEngine.hpp:112]: Unfortunately to declare class member with `auto` it needs to be also `static` - I removed this `int` only to unify unsigned integers declarations across the whole file, as all other declarations here omit it.

Comment (exeosdev) [reply module-gui/gui/widgets/ListViewEngine.hpp:112]: Ah, just noticed the context! Too many things at the same time, sorry :)

**Review by exeosdev — APPROVED** [review]

**Pull request merged**